docs/tutorials/python/sagas/index.md

```diff
@@ -0,0 +1,586 @@
+---
+id: sagas
+sidebar_position: 4
+keywords: [Python, temporal, sdk, tutorial, saga, saga pattern, rollback, compensating actions]
+tags: [Python, SDK]
+last_update:
+  date: 2024-03-06
+title: Build a trip booking application in Python
+description: Implement the Saga Pattern in Python using Temporal.
+image: /img/temporal-logo-twitter-card.png
+---
+
+When dealing with distributed systems, a failure in one service can lead to a domino effect, compromising the entire transaction.
+The Saga pattern offers a solution to this problem by allowing distributed transactions to be broken into smaller, manageable transactions, each with its own compensation logic in case of failure.
+
+The Saga pattern is a design pattern that provides a mechanism to manage long-running transactions and ensure data consistency across multiple services.
+Instead of a single monolithic transaction, the Saga pattern breaks the transaction into smaller, manageable steps (Activities), each step is executed sequentially, and if a step fails, previous steps are undone with a compensating step.
+
+Temporal orchestrates long-running transactions, automatically compensating for failures.
```

Comment:
Two things here: First, I'm not sure if we're still supposed to use "orchestration" (cc @MasonEgger ) and second, it would be a bonus to expand this to explain in a little more depth. There's a lot happening in this sentence.